### ISSUE TITLE
[master] log directory not created in time when running with pre-made configs

### ISSUE THREAD

**Issue by Torxed** [archlinux/archinstall#1865]
![screenshot](https://github.com/archlinux/archinstall/assets/861439/6c5be46f-459f-43fd-9092-9e8d56b9d72e)

**Comment by svartkanin** [comment]
The error here isn't the actual log problem that is just the result of the upper error which is the actual problem `Can't have partition outside disk`

**Comment by Torxed** [comment]
We should probably move the first log instance + creation to the very start of archinstall when we log the sysinfo.

The disk thing was a separate issue that I'm working on while experimenting with `/boot` being two partitions. Got the sectors wrong ^^

**Comment by svartkanin** [comment]
I will yield to @codefiles suggestions to always verify log permissions and the directory. All options that went through my head seemed worse :) 

I raised a PR to address it

**Comment by codefiles** [comment]
I really would not consider that yielding to me. What I have suggested is to verify when writing to the log, which is what was done prior to 89cefb9. To me what you implemented previously with `_check_log_permissions()` and what you are doing in the new pull request appear to be an attempt at optimization. As I demonstrated here https://github.com/archlinux/archinstall/pull/1831#issuecomment-1578603332 the optimization saves virtually no time.
